### PR TITLE
David orders

### PR DESCRIPTION
**Pull request by DavidTTran**
Completed
- User stories 30, 31
- Adjusted some of the routes
   - get `'/profile/orders/:order_id', to: 'order#show'` (maybe this should be `user_order#show`?)
- Added item_order.status that default to 'Unfulfilled'
- User can cancel an order from the show page.
   - Updates order.status to 'Canceled'
   - Assigns item_order.status to 'Unfulfilled'
   - Returns item_order.quantity to item.inventory

### CODE REVIEW

File: spec/features/users/orders_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    click_link("Cancel Order")
+
+    expect(current_path).to eq("/profile")
+    expect(page).to have_content("Your order has been canceled.")
+    expect(order_1.reload.status).to eq("Canceled")
+    expect(item_order_1.reload.status).to eq("Unfulfilled")
+    expect(tire.reload.inventory).to eq(2)
+    expect(pull_toy.reload.inventory).to eq(3)
+  end
```
Comment (DavidTTran) [inline spec/features/users/orders_spec.rb:28]: At the moment I don't have a way to visually double check on the app - but I hope this test is clear enough. Refer back to the instantiated objects.
- item_orders - starts off as 'Fulfilled'
- order_1 - starts off as 'Packaged'
- items - starts off with inventories of 0

Canceling an order should.. Change the status of item_orders, order, and return the item quantity to the merchant.

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -33,4 +34,8 @@ def create
   def order_params
     params.permit(:name, :address, :city, :state, :zip)
   end
+
+  def update_status
+    @order.update(status: "Packaged")
```
Comment (colintalex) [inline app/controllers/orders_controller.rb:39]: I did get rid of Orders controller but we can move this to UserOrders controller where I have already moved everything. See my PR

File: app/controllers/user_orders_controller.rb [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
-    
+  end
+
+  def cancel
+    order = Order.find(params[:order_id])
+    order.update(status: "Canceled")
+    if order.save
+      order.item_orders.each do |item_order|
+        item = Item.find(item_order.item_id)
+        item.inventory += item_order.quantity
```
Comment (colintalex) [inline app/controllers/user_orders_controller.rb:11]: This can stay in this controller

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -6,6 +6,7 @@ def new
 
   def show
     @order = Order.find(params[:id])
+    return update_status if @order.item_orders.all? {|item_order| item_order.status == "Fulfilled"}
```
Comment (Maxwell-Baird) [inline app/controllers/orders_controller.rb:9]: We will need to move this after merging

**Review by colintalex — APPROVED** [review]

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -33,4 +34,8 @@ def create
   def order_params
     params.permit(:name, :address, :city, :state, :zip)
   end
+
+  def update_status
+    @order.update(status: "Packaged")
+  end
```
Comment (Maxwell-Baird) [inline app/controllers/orders_controller.rb:40]: This will need to be moved

File: config/routes.rb [hunk 6]
```diff
@@ -44,10 +43,13 @@
 
   get "/profile", to: "profiles#show"
   get "/profile/orders", to: 'user_orders#index'
+  get "/profile/orders/:id", to: "orders#show"
```
Comment (Maxwell-Baird) [inline config/routes.rb]: We will need route this to user_orders#show after merge

**Pull request closed without merging**